app-backend/datamodel/src/main/scala/AnnotationProjectStatus.scala

```diff
@@ -0,0 +1,65 @@
+package com.rasterfoundry.datamodel
+
+import cats.syntax.either._
+import io.circe._
+
+import java.security.InvalidParameterException
+
+sealed abstract class AnnotationProjectStatus(val repr: String) {
+  override def toString = repr
+
+  /** Order possible statuses to support annotation project workflow
+    *
+    * These values are used to filter to annotation projects with statuses
+    * no greater than a specified value, so for example, projects that
+    * have either failed to be ingested or haven't been attempted but
+    * that aren't queued or currently ingesting.
+    */
+  def toInt: Int = repr.toUpperCase match {
+    case "WAITING"                 => 1
+    case "QUEUED"                  => 2
+    case "PROCESSING"              => 3
+    case "READY"                   => 4
+    case "UNKNOWN_FAILURE"         => 5
+    case "TASK_GRID_FAILURE"       => 6
+    case "IMAGE_INGESTION_FAILURE" => 7
+  }
+}
+
+object AnnotationProjectStatus {
+  case object Waiting extends AnnotationProjectStatus("WAITING")
+  case object Queued extends AnnotationProjectStatus("QUEUED")
+  case object Processing extends AnnotationProjectStatus("PROCESSING")
+  case object Ready extends AnnotationProjectStatus("READY")
+  case object UnknownFailure extends AnnotationProjectStatus("UNKNOWN_FAILURE")
+  case object TaskGridFailure
+      extends AnnotationProjectStatus("TASK_GRID_FAILURE")
+  case object ImageIngestionFailure
+      extends AnnotationProjectStatus("IMAGE_INGESTION_FAILURE")
```

Comment:
I think the data could be modeled a little bit better here. In particular, it's not going to be super ergonomic that the three failure states are just some strings we need to memorize when we're consuming this on the frontend (I know they're enums, but that's not that different). It would be better I think if the data model encoded the differences between "probably fine but not done" and "broken". This would save the future frontend workaround of having a list of error states from otherwise interchangeable members of a type. I don't think the db representation will need to change, but circe codecs and doobie metas can. We can pair Monday so I can explain a bit better if that's ok with you

Reply:
Yea, let's pair for a bit